src/packet.c

```diff
@@ -35,17 +49,23 @@ Packet decode(uint8_t *buf, size_t len, Error *err) {
 
 	packet.header = decode_packet_header(buf);
 
+	memcpy(packet.data, buf + PACKET_HEADER_SIZE, sizeof(packet.data));
```

Comment:
What's the point of storing `data` buffer?

Reply:
In case someone is interested in the compressed payload. But now that I think about it not sure if it is worth it. Could probably removed and we only ever store the decompressed payload.